probdists/Triangulardistribution.py

```diff
@@ -0,0 +1,199 @@
+import math
+import matplotlib.pyplot as plt
+from .Generaldistribution import Distribution
+from collections import Counter
+
+
+class Triangular(Distribution):
+    """
+    Triangular distribution class for calculating and visualizing the
+    triangular distribution: a continuous probability distribution shaped
+    like a triangle
+    Note: a <= mode <= b
+
+    Attributes:
+
+        a (float): the minimum lower limit value
+        b (float): the maximum upper limit value
+        mode (float): the mode, where min <= mode <= max
+
+        mean (float): the mean value of the distribution
+        stdev (float): the standard deviation of the distribution
+
+    """
+
+    def __init__(self, a=0, b=1, mode=0.5):
+        if mode < a or b < mode:
+            raise ValueError
+
+        self.a = a
+        self.b = b
+        self.mode = mode
+
+        Distribution.__init__(self, self.calculate_mean(),
+                              self.calculate_stdev())
+
+    def calculate_mean(self, round_to=2):
+        """
+        Method to calculate the mean from the min, max and mode
+
+        Args:
+            round_to (int): Round the mean value. Defaults to 2.
+
+        Returns:
+            float: mean of the data set
+        """
+
+        self.mean = 1/3 * (self.a + self.b + self.mode)
+
+        return round(self.mean, round_to)
+
+    def calculate_stdev(self, round_to=2):
+        """
+        Method to calculate the standard deviation from the min, max and mode
+
+        Args:
+            round_to (int): Round the mean value. Defaults to 2.
+
+        Returns:
+            float: standard deviation of the data set
+        """
+
+        summation = (self.a ** 2) + (self.b ** 2) + (self.mode ** 2) - \
+                    (self.a * self.b) - (self.a * self.mode) - \
+                    (self.b * self.mode)
+        variance = summation / 18
+        self.stdev = math.sqrt(variance)
+
+        return round(self.stdev, round_to)
+
+    def replace_stats_with_data(self):
+        """ Method to calculate a, b, mode from the data set
+
+        Args:
+            None
+
+        Returns:
+            float: a, the minimum value
+            float: b, the maximum value
+            float: mode, the mode of the dataset
+        """
+        if not self.data:
+            # Use default values
+            self.a, self.b, self.mode = 0, 1, 0.5
+        else:
+            self.a = min(self.data)
+            self.b = max(self.data)
```

Comment:
Maybe find the min and max using a single loop instead of iterating over the data twice, because this can be problematic with large data.